File: source_file.py
import socket

HEADER = 64
PORT = 6011
SERVER = 'localhost'
ADDR = (SERVER, PORT)
FORMAT = 'utf-8'
DISCONNECT_MESSAGE = b'!DISSCONNECT'

CONN = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
CONN.connect(ADDR)

def display_msg(msg):
    print(
        '\033[93m', 
        msg, 
        sep='', 
        end='\033[0m\n'
    )

def send_msg(conn, msg):
	data = bytes(msg, FORMAT)
	data += b' ' * (HEADER - len(data))
	conn.send(data)

def recv_msg(conn):
	msg = conn.recv(HEADER).strip()
	msg = msg.decode(FORMAT)
	return msg

msg = "Initializing Client - Python"
display_msg(msg)

msg = "Initialized Client - Python"
send_msg(CONN, msg)

while msg != '!DISCONNECT':
	msg = recv_msg(CONN)
	display_msg(msg)

CONN.close()

File: source_file.r
HEADER <- 64
PORT <- 6011
SERVER <- "localhost"
FORMAT <- "utf-8"
DISCONNECT_MESSAGE <- "!DISSCONNECT"

con <- socketConnection(
	host = SERVER,
	port = PORT,
	server = FALSE
)

display_msg <- function(msg) {
    cat(
        '\033[94m',
        msg,
        sep = '',
        end = '\033[0m\n'
    )
}

send_msg <- function(conn, msg) {
	sendme <- paste(msg, strrep(" ", HEADER - nchar(msg)), sep = "")
	writeChar(sendme, conn)
}

recv_msg <- function(conn) {
	suppressWarnings(msg <- trimws(readChar(conn, HEADER)))
	while (length(msg) == 0) {
		#cat('ALERT\n')
		suppressWarnings(msg <- trimws(readChar(conn, HEADER)))
	}
	return(msg)
}

msg <- "Initializing Client - R"
display_msg(msg)

msg <- "Initialized Client - R"
send_msg(con, msg)

while (msg != '!DISCONNECT') {
	msg <- recv_msg(con)
	display_msg(msg)
}

close(con)